examples/mnist/README.md

```diff
@@ -34,32 +34,55 @@ class ModelOwner:
 class PredictionClient:
 
     def provide_input(self) -> tf.Tensor:
-        prediction_input = self._build_data_pipeline().get_next()
-        # pre-processing
-        prediction_input = tf.reshape(prediction_input, shape=(self.BATCH_SIZE, 28 * 28))
+        """Prepare input data for prediction."""
+        with tf.name_scope('loading'):
+        prediction_input, expected_result = self._build_data_pipeline().get_next()
+        print_op = tf.print("Expect", expected_result, summarize=self.BATCH_SIZE)
+        with tf.control_dependencies([print_op]):
+            prediction_input = tf.identity(prediction_input)
+
+        with tf.name_scope('pre-processing'):
+        prediction_input = tf.reshape(
+            prediction_input, shape=(self.BATCH_SIZE, ModelOwner.FLATTENED_DIM))
         return prediction_input
 
-    def receive_output(self, likelihoods: tf.Tensor) -> tf.Operation:
-        # post-processing
-        prediction = tf.argmax(likelihoods, axis=1)
+    def receive_output(self, logits: tf.Tensor) -> tf.Operation:
+        with tf.name_scope('post-processing'):
+        prediction = tf.argmax(logits, axis=1)
         op = tf.print("Result", prediction, summarize=self.BATCH_SIZE)
         return op
 ```
 
 Instances of these are then linked together in a secure computation performing a prediction, treating both the weights and the prediction input as private values.
 
 ```python
-model_owner = ModelOwner('model-owner')
-prediction_client = PredictionClient('prediction-client')
-
-w0, b0, w1, b1 = tfe.define_private_input(model_owner.player_name, model_owner.provide_input)
-x = tfe.define_private_input(prediction_client.player_name, prediction_client.provide_input)
-
-layer0 = tfe.matmul(x, w0) + b0
-layer1 = tfe.sigmoid(layer0)
-logits = tfe.matmul(layer1, w1) + b1
-
-prediction_op = tfe.define_output(prediction_client.player_name, logits, prediction_client.receive_output)
+  model_owner = ModelOwner(
+      player_name="model-owner",
+      local_data_file="./data/train.tfrecord")
+
+  prediction_client = PredictionClient(
+      player_name="prediction-client",
+      local_data_file="./data/test.tfrecord")
+
+  with tfe.protocol.SecureNN():
+    batch_size = PredictionClient.BATCH_SIZE
+    flat_dim = ModelOwner.IMG_ROWS * ModelOwner.IMG_COLS
+    batch_input_shape = [batch_size, flat_dim]
+
+    model = tfe.keras.Sequential()
```

Comment:
we don't show that the model owner provides the model weights 
```
params = tfe.define_private_input(model_owner.player_name,
                                    model_owner.provide_input)
```

we should also add `model.set_weights(params, sess)` after the last model.add